prql-compiler/tests/docs/main.rs

```diff
@@ -0,0 +1,75 @@
+use prql_compiler::Options;
+use regex::Regex;
+use serde_yaml::Value;
+use std::fs;
+
+const WEBSITE_TOPPAGE: &str = "../web/website/content/_index.md";
+
+fn compile(prql: &str) -> Result<String, prql_compiler::ErrorMessages> {
+    prql_compiler::compile(prql, &Options::default().no_signature())
+}
+
+fn sql_normalize(sql: &str) -> String {
+    let re = Regex::new(r"\n\s+").unwrap();
+    re.replace_all(sql, " ").trim().to_string()
+}
+
+fn website_contents() -> Value {
+    let contents = fs::read_to_string(WEBSITE_TOPPAGE)
+        .unwrap()
+        .replace("---", "");
+
+    serde_yaml::from_str::<Value>(&contents).unwrap()
+}
+
+fn website_examples() -> Vec<Value> {
+    let value = website_contents();
+
+    value
+        .get("showcase_section")
+        .unwrap()
+        .get("examples")
+        .unwrap()
+        .as_sequence()
+        .unwrap()
+        .to_vec()
+}
+
+fn website_hero_example() -> String {
+    let value = website_contents();
+
+    value
+        .get("hero_section")
+        .unwrap()
+        .get("prql_example")
+        .unwrap()
+        .as_str()
+        .unwrap()
+        .to_string()
+}
+
+#[test]
+fn test_readme_examples() {
+    let contents = fs::read_to_string("../README.md").unwrap();
```

Comment:
I had thought that these lines were somehow doing this: https://github.com/PRQL/prql/blob/99b82a3afb3deda0168ba4d79226574e0fab8bed/prql-compiler/src/lib.rs#L244-L246

But I guess they aren't... (I'm not even sure how it would be parsing the PRQL. But then I'm also not sure what that coding is doing then...)

Reply:
I am hardly familiar with the Rust documentation, but perhaps this is a feature for testing Rust code blocks?

Reply:
Ah sorry, that's pulling in the `prql-compiler/README.md` and testing that rust!